src/openrct2/ride/TrackDesign.cpp

```diff
@@ -977,23 +976,30 @@ static bool TrackDesignPlaceSceneryElementGetPlaceZ(const TrackDesignSceneryElem
     return true;
 }
 
-static bool TrackDesignPlaceSceneryElement(
-    CoordsXY mapCoord, uint8_t mode, const TrackDesignSceneryElement& scenery, uint8_t rotation, int32_t originZ)
+static std::optional<money32> TrackDesignPlaceSceneryElement(
```

Comment:
I feel this is pretty odd way of doing these functions. It would be more logical that the std::optional<money32> is just the result of placing this singular piece of scenery. Move the actual addition of totalCost to the caller. Then we can just ignore everything to do with `MONEY32_UNDEFINED`. The other alternative is to not use std::optional and only return money32 and use MONEY32_UNDEFINED to indicate failure.